jplag/src/main/java/de/jplag/options/JPlagOptions.java

```diff
@@ -218,11 +219,13 @@ public void setRootDirectoryName(String rootDirectoryName) {
     }
 
     public void setBaseCodeSubmissionName(String baseCodeSubmissionName) {
-        this.baseCodeSubmissionName = baseCodeSubmissionName;
+        // Trim problematic file separators.
+        this.baseCodeSubmissionName = (baseCodeSubmissionName == null) ? null : baseCodeSubmissionName.replace(File.separator, "");
     }
 
     public void setSubdirectoryName(String subdirectoryName) {
-        this.subdirectoryName = subdirectoryName;
+        // Trim problematic file separators.
+        this.subdirectoryName = (subdirectoryName == null) ? null : subdirectoryName.replace(File.separator, "");
     }
```

Comment:
Now that I look at that code I wonder if settings null here should even be allowed? If not, we should check for null and throw an `IllegalArgumentException` if null. If not, we can safely trim here without the not to nice `(subdirectoryName == null) ? null` check.

Reply:
In which cases did you previously encounter an NPE here? What caused it?

Reply:
One of the test cases crashed.

Afaik `subdirectoryName` is an optional name of a folder inside a submission containing the source code (`root/*` for submissions versus `root/*/subdir`). Being optional, I'd expect it to be `null` sometimes (which seems to be the case for its definition ` SUBDIRECTORY("-S", String.class),`, there is no default value.

Reply:
Hmm, actually I fixed the crash on the base code and then used the above idea to conclude that sub directory needed it as well.

Reply:
Ah, I see, the CLI will set all options, since argparse4j will return null if an argument is not set.
We could change the CLI to only call the setter if the value is non-null, however for now we can leave it like this.